cpu/stm32/periph/pwm.c

```diff
@@ -53,7 +53,7 @@ uint32_t pwm_init(pwm_t pwm, pwm_mode_t mode, uint32_t freq, uint16_t res)
     dev(pwm)->CR1 = 0;
     dev(pwm)->CR2 = 0;
     for (unsigned i = 0; i < TIMER_CHAN; ++i) {
-        dev(pwm)->CCR[i] = 0;
+        *(__IO uint32_t *)(&dev(pwm)->CCR1 + i) = 0;
```

Comment:
I'm not sure if the case to volatile is needed here, you are declaring the pointer as volatile, when I think only the content should be volatile, the address (pointer) will not change. (same for other occurences)
```suggestion
        *(&dev(pwm)->CCR1 + i) = 0;
```

Reply:
(I'm very unsure about this)

Reply:
Good point. It seems to work and the code size is smaller (only 4B on ROM added now)